ruff_cli/src/printer.rs

```diff
@@ -148,6 +148,27 @@ impl<'a> Printer<'a> {
             }
             return Ok(());
         }
+        // Prints a count of the number of violations for each rule if statistics are
+        // enabled.
+        if matches!(self.log_level, LogLevel::Statistics) {
+            let mut stdout = BufWriter::new(io::stdout().lock());
+            let mut violations = diagnostics
+                .messages
+                .iter()
+                .map(|message| message.kind.rule())
+                .collect::<Vec<_>>();
+            violations.sort();
+            violations.dedup();
```

Comment:
Rather than putting this in printer, and introducing a new log level... what if, instead, we forked where we call `printer.write_once(&diagnostics)?` from `main.rs`. That is: check if we're doing statistics, then call a new `Printer::statistics` instead of `Printer::write_once`?

That way, we can also have different formats for statistics. For example, we could output JSON! And the implementation would be entirely separate from this method, which deals with a different kind of task and is already a little complex.

Reply:
good idea! on it!